### PR TITLE
Add role columns tables 1550

### PR DESCRIPTION
**Pull request by gordonfarrell**
## Background
Setup new user role table and create relationships between roles and permissions. My first time using typeorm so please take a close look and make sure I don't bork anything with these migrations. Thanks!

## GitHub Issue
[#1550](https://github.com/ctoec/data-collection/issues/1550)

## Validation Plan
Migrations seem to run properly for me, so confirmation from others that the DB is ending up in the state we want it should do it.

## Tech Specs
- [X] If any TypeORM entities were created/updated/removed, the necessary migration(s) were [correctly auto-generated](https://github.com/typeorm/typeorm/blob/master/docs/using-cli.md#generate-a-migration-from-existing-table-schema) and **not manually written**.

## Automated Testing
- [X] All unit tests are passing.
- [N/A] Unit tests have been added to cover this changeset.

### CODE REVIEW

File: client/src/shared/models/RoleEnum.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export enum RoleEnum {
```
Comment (lukenewman) [inline client/src/shared/models/RoleEnum.ts:1]: can we call this just `Role` instead, or `RoleType`?

**Review by lukenewman — COMMENTED** [review]
I'm a bit confused on the technical specs of this ticket. Can't we just have a `RoleType` (currently `RoleEnum`) relationship on the `User`? I don't understand why the `Role` table is required with an `id` and all...

File: client/src/shared/models/RoleEnum.ts (hunk 1 above)
Comment (gordonfarrell) [reply client/src/shared/models/RoleEnum.ts:1]: Sure, renaming sounds good, though I'll change it to roleType. I tried using just role as the name and my IDE was unhappy with it, I think just `role` may be a reserved word

**Review by lukenewman — APPROVED** [review]

**Review by jessethach — APPROVED** [review]

**Comment by jessethach** [comment]
I can validate this PR.

**Comment by jessethach** [comment]
I have validated using the steps below:

**Log in with admin/provider level user**

- [ ] Successfully log into app
- [ ] Check db for new `user_role_role` table. Expected to be empty.
- [ ] Check db permissions tables to contain `roleid` column.

**Comment by bamader** [comment]
@jessethach looks like you have checks for all your tests. Does this PR pass validation in your opinion? (this is just something we do for history/documentation purposes--when satisfied, the validator leaves a comment that the PR is good to merge and tags either Nick or me to let us know we can merge to base).

**Comment by jessethach** [comment]
@bamader Good to know! I will do that moving forward. This is ready to merge!

**Pull request merged**